code/game/objects/structures/doors.dm

```diff
@@ -0,0 +1,902 @@
+///Can attach a lock
+#define DOOR_LOCKABLE 1<<0
+///Can be opened with violence
+#define DOOR_FORCEABLE 1<<1
+///If the door is open
+#define DOOR_OPEN 1<<2
+///Is currently opening
+#define DOOR_OPENING 1<<3
+///Is currently closing
+#define DOOR_CLOSING 1<<4
+
+///Set of flags your average door will most likely have
+#define STANDARD_DOOR_FLAGS DOOR_LOCKABLE|DOOR_FORCEABLE
+
+/obj/structure/door
+	name = "door"
+	desc = "It opens and closes - nothing out of the ordinary."
+	icon = 'icons/obj/doors/mineral_doors.dmi'
+	icon_state = ""
+	opacity = TRUE
+	density = TRUE
+	allow_pass_flags = PASS_AIR
+	resistance_flags = XENO_DAMAGEABLE
+	layer = DOOR_CLOSED_LAYER
+	max_integrity = 100
+	explosion_block = 0.5
+	///The name of the door, assigned at initialization; use this instead of initial(name)
+	var/base_name
+	///Door specific flags
+	var/door_flags = STANDARD_DOOR_FLAGS
+	///Reference to the lock installed on this door
+	var/obj/item/lock/lock
+	///What material the door is made of; used for repairs and deconstruction
+	var/obj/item/stack/material_type = /obj/item/stack/sheet/wood
+	///How many units of material this door is made of
+	var/material_amount = 4
+	///Sound when opened
+	var/open_sound = 'sound/machines/door_open.ogg'
+	///Sound when closed
+	var/close_sound = 'sound/machines/door_close.ogg'
+	///Sound when locked and trying to open
+	var/locked_sound = 'sound/machines/door_locked.ogg'
+	///Sound when trying to use violence against a door
+	var/slam_sound = 'sound/effects/slam3.ogg'
+	///Sound when knocking on a door
+	var/knocking_sound = 'sound/effects/glassknock.ogg'
+	///How long it takes to open the door; recommended to not make this the entire length of the animation for "feel" reasons
+	var/opening_time = 0.5 SECONDS
+	///How long it takes to close the door; will be the same as opening_time if not set
+	var/closing_time
+	///Reference to the timer that closes or opens a door
+	var/delay_timer
+	///Steps in sequential order to de/construct this door; make this list null if you do not want this door to be built or taken apart
+	var/list/construction_steps = list(TOOL_SCREWDRIVER, TOOL_CROWBAR)
+	///What step in the de/construction process this door is on; 1 means it is fully built
+	var/construction_steps_index = 1
```

Comment:
It's kind of the same, still strings under these defines. I'm basing it off of the tool defines because I think it's easier for a step-by-step list.

"First the wrench, then the crowbar, then the welder."

All the de/construction code is check if the tool type matches the one in the list at the index.